automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/AsyncTestUtils.scala

```diff
@@ -0,0 +1,79 @@
+package org.broadinstitute.dsde.test.api.orch
+
+import org.scalatest.Assertion
+import org.scalatest.concurrent.Eventually
+import org.scalatest.concurrent.PatienceConfiguration.{Interval, Timeout}
+import org.scalatest.exceptions.TestFailedException
+
+import javax.xml.parsers.FactoryConfigurationError
+
+/**
+  * Some helper functions to get asynchronous tests failing quickly
+  * Should some day be made available to a wider codebase
+  */
+trait AsyncTestUtils extends Eventually {
+
+  /**
+    * Wraps a function in an eventually wrapped in a try-catch;
+    * Functions as an eventually but will fail test without retrying if test-ending errors thrown in the test function
+    * @param timeout
+    * @param interval
+    * @param testFunction Test to run; should include a failAndEscape (could be within assertWithEscape)
+    * @tparam T the result of the testFunction
+    * @return The result of the testFunction (an assertion) if an error is not thrown
+    */
+  def eventuallyWithEscape[T](timeout: Timeout, interval: Interval, testFunction: => T): T =
```

Comment:
nit: I'd vote to put the timeout/interval arguments second, and then make them optional. Something like:
`def eventuallyWithEscape[T](testFunction: => T, timeout: Timout=None, interval=None): T =`
And then combine the two functions. Maybe there's an even fancier technique out there to mirror the eventually functions? (Is it possible to have a curried function where the first parameter is either the timeout or the function? I think this would be more work than it's worth)

Reply:
In the original eventually they're implicit https://github.com/scalatest/scalatest/blob/7aeda262e249a66d2319b6b2f5daa1b7d0cf9a4f/jvm/core/src/main/scala/org/scalatest/concurrent/Eventually.scala#L414
so that if you don't supply them it grabs them from PatienceConfig.  So I was relying on the `eventually` inside `eventuallyWithEscape` to take care of this in the second function that only has one parameter.  I don't quite understand how implicit works so I didn't want to try doing that myself, but I'm happy to rewrite/rearrange these functions however works best.

Reply:
I see. I would still vote for optional arguments in that case, to reduce duplication. We could just call eventually with the timeout arguments if they're not none, but leave them out otherwise.

Reply:
I think the best thing to do here is to copy the signature of the `eventually` that you're wrapping, which is [this one](https://github.com/scalatest/scalatest/blob/7aeda262e249a66d2319b6b2f5daa1b7d0cf9a4f/jvm/core/src/main/scala/org/scalatest/concurrent/Eventually.scala#L312). Since its signature is
```scala
def eventually[T](timeout: Timeout, interval: Interval)(fun: => T)(implicit retrying: Retrying[T], pos: source.Position): T
```
you'd have:
```scala
def eventuallyWithEscape[T](timeout: Timeout, interval: Interval)(testFunction: => T)(implicit retrying: Retrying[T], pos: source.Position): T
```
( do the same for the `eventuallyWithEscape` at line 40 too )

FYI, IntelliJ's `View -> Show Implicit Hints` and sometimes `View -> Expand Implicit Hints` can be helpful to follow all the sneaky implicits through! Mostly I hate implicits.